packages/table/components/Table.tsx

```diff
@@ -0,0 +1,172 @@
+import * as React from "react";
+import { css } from "emotion";
+import { AutoSizer, MultiGrid } from "react-virtualized";
+import { IColumnProps, Column } from "./Column";
+import memoizeOne from "memoize-one";
+import { coreColors as Colors } from "../../shared/styles/color";
+
+export interface ITableProps {
+  /**
+   * data is expecting an array of any values. These values should contain all the data represented in the table.
+   */
+  data: any[];
+  /**
+   * children can only be of type Column
+   */
+  children:
+    | Array<React.ReactElement<IColumnProps>>
+    | React.ReactElement<IColumnProps>;
+}
+
+const rowHeight = 35;
+export class Table<T> extends React.PureComponent<ITableProps, {}> {
+  public multiGridRef: { recomputeGridSize?: any } = {};
+
+  public getData = memoizeOne(
+    (data: T[]): Array<{}> | T[] => {
+      return [{}, ...data];
+    }
+  );
+
+  public getColumnSizes = memoizeOne(
+    (
+      children: Array<React.ReactElement<IColumnProps>>,
+      width: number
+    ): number[] => {
+      const totalColumns: number = children.length;
+      let remainingWidth: number = width;
+      return children.map((child, currentIndex) => {
+        const calculatedWidth = child.props.width({
+          width,
+          totalColumns,
+          currentIndex,
+          remainingWidth
+        });
+        remainingWidth -= calculatedWidth;
+        return calculatedWidth;
+      });
+    }
+  );
+
+  constructor(props) {
+    super(props);
+
+    this.getCell = this.getCell.bind(this);
+    this.setRef = this.setRef.bind(this);
+    this.updateGridSize = this.updateGridSize.bind(this);
+  }
+
+  public render() {
+    const mergedData = this.getData(this.props.data);
+
+    const columnCount = React.Children.count(this.props.children);
+
+    return (
+      <AutoSizer
+        defaultWidth={1024}
+        defaultHeight={768}
+        onResize={this.updateGridSize}
+      >
+        {({ width, height }) => {
```

Comment:
We may wanna avoid arrow functions in JSX as they affect performance. Why not extracting a component?

Reply:
Let me try that. You are right. But the performance reason does not fit here. Since the lambda function is not passed as properties but as children. This is also how it is solved in the docs: https://github.com/bvaughn/react-virtualized/blob/master/docs/AutoSizer.md